### PR TITLE
switch GCR -> Artifact-Registry

### PR DESCRIPTION
**Pull request by ccwienk**
GCR has been deprecated [0] in favour of Artifact-Registry.

Thus, change push-targets for OCI-Images:

- europe-docker.pkg.dev/gardener-project/snapshots for snapshots
- europe-docker.pkg.dev/gardener-project/releases for releases
- europe-docker.pkg.dev/gardener-project/public for combined view of snapshots + releases

[0]
https://cloud.google.com/artifact-registry/docs/transition/transition-from-gcr

**Release note**:

```breaking operator
Change OCI Image Registry from GCR (`eu.gcr.io/gardener-project`) to Artifact-Registry (`europe-docker.pkg.dev/gardener-project/releases`). Users should update their references.

```

### CODE REVIEW

File: publishing-cfg.yaml [hunk 1]
```diff
@@ -76,7 +76,7 @@
       bucket_name: 'gardenlinux-github-releases'
       aws_cfg_name: 'gardenlinux-integration-test'
   ocm:
-    component_repository_cfg_name: 'gardenlinux-test'
```
Comment (MrBatschner) [inline publishing-cfg.yaml:79]: `gardenlinux-test` pointed to repository `eu.gcr.io/gardenlinux/integration-tests` - the cds created by this publishing config should not live in the `gardener-project` universe.

File: .cicd-cli.py [hunk 2]
```diff
@@ -975,8 +975,7 @@ def cleanup_release_set():
     cfg_factory = ctx.cfg_factory()
 
     if not parsed.ocm_repo:
-        ocm_cfg = cfg_factory.ctx_repository(cfg.ocm.component_repository_cfg_name)
```
Comment (MrBatschner) [inline .cicd-cli.py:978]: Just for clarification: before, `cfg_factory.ctx_repository()` took the name of a config item and resolved that to a repo context and now it takes the base-url of a repository and just returns that url?

**Review by MrBatschner — CHANGES_REQUESTED** [review]

File: publishing-cfg.yaml (hunk 1 above)
Comment (ccwienk) [reply publishing-cfg.yaml:79]: valid point. in that case, I think we will need an artifact-registry for `gcp/gardenlinux`.

File: .cicd-cli.py (hunk 2 above)
Comment (ccwienk) [reply .cicd-cli.py:978]: we might reconsider whether a cfg-element is actually adequate in this special case

File: publishing-cfg.yaml [hunk 3]
```diff
@@ -76,7 +76,7 @@
...
+    ocm_repository: europe-docker.pkg.dev/gardener-project/snapshots
```
Comment (MrBatschner) [inline publishing-cfg.yaml]:
```suggestion
    ocm_repository: europe-docker.pkg.dev/sap-se-gcp-gardenlinux/tests
```

**Review by MrBatschner — CHANGES_REQUESTED** [review]
Pasted the ocm_repository for GL tests as suggestion inline.
LGTM otherwise.

**Review by MrBatschner — APPROVED** [review]
/lgtm

**Pull request merged**